docs/proposals/operator-managed-infra.md

```diff
@@ -0,0 +1,116 @@
+* Make install onto existing K8s cluster straight forward and remove
+  complexity of using Terraform
+* Improved UX, all an end-user needs to do is install our manifests and ensure GCP service account
+  has enough permissions
+* Enterprises generally have very specific requirements
+  about how they create their K8s Clusters and nodepools.
+* Allow Substratus to work in a multi namespace environment.
+  Currently namespace is hardcoded in workload identity settings in the terraform
+
+## How?
+* Make the Image Registry and Object Storage Bucket configurable as a setting of the operator itself
+* Object storage will be optional and if not provided then default PVC class will be used
+* Utilize a single service account that has enough permissions to do everything needed within Substratus
+* Simplify Substratus to use a single K8s SA per namespace. Simplicity is preferred here over the minor security benefits you get
+  for using a different SA for each kind of Substratus resource.
+* Provide Substratus controller the permissions to manage registry, bucket and set IAM policy on the GCP SA.
+  The following permissions will be required by the substratus Service Account on the project itself:
+  ```
+  roles/storage.admin, roles/artifactregistry.repoAdmin, roles/iam.serviceAccountAdmin
+  ```
+
+### Implementation details
+* Object storage: Create bucket if the configured bucket does not exist
+* Image Registry: Create registry if the configured registry does not exist
+* New sub reconciler for namespace: Create K8s Service Accounts automatically in each namespace where Substratus is used and
+  annotate the K8s Service Account with the correct GCP Service Account. Ensure that following
+  gets called when a new Service Account is provisioned in a new namespace:
+  ```
+  gcloud iam service-accounts add-iam-policy-binding substratus@my-project.iam.gserviceaccount.com \
+   --role roles/iam.workloadIdentityUser \
+   --member "serviceAccount:myproject.svc.id.goog[new-namespace/substratus]"
+  ```
+  Make sure this only happens whenever a new Substratus gets created in a new namespace.
+  The controller should be smart enough to not have to create KSAs in namespaces that
+  do not have any substratus resource.
+
+How should Substratus controller handle infra management?
+* Terraform
+* Native K8s controller that uses Golang SDK for AWS/GCP/..
+
+The Native K8s controller would better fit within the existing Substratus code base.
+It would also allow us to verify if a resource is already there and if not just create
+it. Terraform would still be used for creation of GKE/EKS cluster to provide a reference
+example for easy install.
+
+### Open Questions
+
+1. Where should the code to create Image Registry and Bucket live? Do we need a new
+   resource or can we reconcile based purely on configuration?
+2. Should Substratus install and manage Karpenter on AWS?
```

Comment:
I believe we landed on no for this and similar cluster-level DaemonSets (e.g., nvidia drivers) though cluster configurations like this would be good to verify through the controller. If incompatible configurations are found, the controller should likely emit warning level logs.